.github/workflows/repo-metrics-weekly.yml

```diff
@@ -0,0 +1,58 @@
+name: Weekly repo metrics
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 9 * * MON'
+
+permissions:
+  issues: write
+  pull-requests: read
+
+jobs:
+  build:
+    # this workflow will always fail in forks; bail if this isn't running in the upstream
+    if: github.repository == 'aws/aws-cdk'
+    name: metrics
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Get dates for last week
+      shell: bash
+      run: |
+        # Calculate the previous Monday's date
+        PREVIOUS_MONDAY=$(date -d "last monday" "+%Y-%m-%d")
+
+        # Calculate the current Sunday's date
+        CURRENT_SUNDAY=$(date -d "current Sunday" "+%Y-%m-%d")
```

Comment:
Just wondering if this command returns the correct previous monday and current sunday? Because to me seems both identical `+%Y-%m-%d`. Do you have any reference doc or any sample?

Reply:
nit: consider renaming `last monday` to `previous monday` or vice versa to be consistent.

Reply:
Oh wait you're right that's not actually doing anything right now, let me have a look

Reply:
Pushed updated shell script. Here's some docs to reference: https://ioflood.com/blog/bash-date-format/